src/OpenTelemetry.ResourceDetectors.Azure/AzureVMResourceDetector.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="AzureVMResourceDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.ResourceDetectors.Azure;
+
+/// <summary>
+/// Resource detector for Azure VM environment.
+/// </summary>
+public class AzureVMResourceDetector : IResourceDetector
+{
+    internal static readonly IReadOnlyCollection<string> ExpectedAzureAmsFields = new string[]
+    {
+        "azInst_location",
```

Comment:
Shall we introduce a AzureConstants or AzureConventions to group these string.

Reply:
Yes, let me do that as follow up. Would do it for AppService too.